src/v/kafka/server/group_manager.cc

```diff
@@ -551,15 +553,16 @@ group_manager::join_group(join_group_request&& r) {
         vlog(klog.trace, "Created new group {} while joining", r.data.group_id);
     }
 
-    return group->handle_join_group(std::move(r), is_new_group)
-      .finally([group] {});
+    auto ret = group->handle_join_group(std::move(r), is_new_group);
+    return group::join_group_stages(
+      std::move(ret.dispatched), ret.result.finally([group] {}));
```

Comment:
why doesn't `ret.dispatched` also keep `group` alive? it seems like the caller could choose to throw away ret.result (and the reference keeping group alive) and then interact with ret.dispatched with UAF bug?

Reply:
`ret.result` will always resolve after the `dispatched`

Reply:
@travisdowns did you file a ticket for this so we don't lose track?